src/deadline/job_attachments/asset_sync.py

```diff
@@ -105,6 +126,333 @@ def __init__(
+
+        if manifest_properties.fileSystemLocationName:
+            local_destination = storage_profiles_path_mapping_rules.get(root_path)
+        else:
+            path_mapping: Optional[PathMappingRule] = dynamic_mapping_rules.get(root_path)
+            local_destination = path_mapping.destination_path if path_mapping else None
+
+        if local_destination:
+            return local_destination
+        else:
+            raise AssetSyncError(
+                "Error occurred while attempting to sync input files: "
+                f"No path mapping rule found for the source path {manifest_properties.rootPath}"
+            )
+
+    def aggregate_asset_root_manifests(
+        self,
+        session_dir: Path,
+        s3_settings: JobAttachmentS3Settings,
+        queue_id: str,
+        job_id: str,
+        attachments: Attachments,
+        step_dependencies: Optional[list[str]] = None,
+        dynamic_mapping_rules: dict[str, PathMappingRule] = {},
+        storage_profiles_path_mapping_rules: dict[str, str] = {},
+    ) -> DefaultDict[str, list[BaseAssetManifest]]:
+        """
+         Args:
+            session_dir: the directory that the session is going to use.
+            s3_settings: S3-specific Job Attachment settings.
+            queue_id:    the ID of the queue for step-step dependency.
+            job_id:      the ID of the job for step-step dependency.
+            attachments: an object that holds all input assets for the job.
+            step_dependencies: the list of Step IDs whose output should be downloaded over the input job attachments.
+            dynamic_mapping_rules: manifest root path to worker host destination mapping relative to local session.
+            storage_profiles_path_mapping_rules: manifest root path to worker host destination mapping given storage profile.
+        Returns: a dictionary of manifest files stored in the session directory.
+        """
+        grouped_manifests_by_root: DefaultDict[str, list[BaseAssetManifest]] = DefaultDict(list)
+
+        for manifest_properties in attachments.manifests:
+            local_root: str = AssetSync.get_local_destination(
+                manifest_properties=manifest_properties,
+                dynamic_mapping_rules=dynamic_mapping_rules,
+                storage_profiles_path_mapping_rules=storage_profiles_path_mapping_rules,
+            )
+
+            if manifest_properties.inputManifestPath:
+                manifest_s3_key = s3_settings.add_root_and_manifest_folder_prefix(
+                    manifest_properties.inputManifestPath
+                )
+                manifest = get_manifest_from_s3(
+                    manifest_key=manifest_s3_key,
+                    s3_bucket=s3_settings.s3BucketName,
+                    session=self.session,
+                )
+                grouped_manifests_by_root[local_root].append(manifest)
+
+        # Handle step-step dependencies.
+        if step_dependencies:
```

Comment:
Can we make this a shared function? This has good usage for the manifest download API too.

Reply:
Emm I'd take this as a follow-up, may need some more information to understand the use cases.

Reply:
yup ok